scripts/popup.js

```diff
@@ -1,217 +1,323 @@
-const extensionDetected = [];
-const SINGLE_FILE_CORE_EXT_ID = 'jemlklgaibiijojffihnhieihhagocma';
-
-function detectExtension(extensionId, callback) {
-  let img;
-  if (extensionDetected[extensionId]) {
-    callback(true);
-  } else {
-    img = new Image();
-    img.src = `chrome-extension://${extensionId}/resources/icon_16.png`;
-    img.onload = () => {
-      extensionDetected[extensionId] = true;
-      callback(true);
-    };
-    img.onerror = () => {
-      extensionDetected[extensionId] = false;
-      callback(false);
+$(window).ready(() => {
+  const Archivist = {
+    metadataFields: [],
+  };
+
+  /* region Post */
+  function handlePostSuccess(data, status) {
+    console.log(status);
+    console.log(data);
+  }
+
+  function getFormData() {
+    const formElements = $('.metadata_item input');
+
+    const formData = {};
+    // create object with name, type, value, group=generic
+
+    formElements.each((element) => {
+      const curElement = formElements[element];
+      formData[curElement.getAttribute('name')] = curElement.value;
+    });
+
+    return formData;
+  }
+
+  function postDataToApi(blob) {
+    const formData = getFormData();
+
+    // loop through each metadataField, add the value from form data to expected API format
+    Archivist.metadataFields.forEach((field) => {
+      field.data = formData[field.name];
+    });
+
+    const tagsData = $('#tags').val();
```

Comment:
Since it all gets turned into a string I didn't think it mattered. The tags are separated by commas in the field correct? Or would you prefer a different deliminator instead and to split on it?